### PR TITLE
[DOP-20048] Split lineage endpoint

### PR DESCRIPTION
**Pull request by TiGrib**
## Change Summary

Split `/v1/lineage` into four different endpoints, for each entity (dataset, operation, run, job)

Remove query class from pydantic request schemas for lineage (https://github.com/fastapi/fastapi/releases/tag/0.115.0).

## Related issue number

[DOP-20059]

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/data-rentgen/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: data_rentgen/server/schemas/v1/lineage.py [hunk 1]
```diff
@@ -96,21 +77,21 @@ def _check_until(cls, value: datetime | None, info: ValidationInfo) -> datetime
...
-            raise ValueError(f"'point_id' should be int for '{self.point_kind}' kind")
-        elif self.point_kind in {LineageEntityKindV1.RUN, LineageEntityKindV1.OPERATION} and not isinstance(
-            self.point_id,
-            OLD_UUID,
-        ):
-            raise ValueError(f"'point_id' should be UUIDv7 for '{self.point_kind}' kind")
-        return self
+
+class DatasetLineageQueryV1(BaseLineageQueryV1):
+    point_id: int = Field(description="Dataset id", examples=[42])
```
Comment (dolfinus) [inline data_rentgen/server/schemas/v1/lineage.py:82]: What do you think about renaming `point_id` -> `start_at_id`?

File: tests/test_server/test_lineage/test_get_lineage_request_validators.py [hunk 2]
```diff
@@ -182,11 +168,10 @@ async def test_get_lineage_until_less_than_since(test_client: AsyncClient):
     until = since - timedelta(days=1)
 
     response = await test_client.get(
-        "v1/lineage",
+        f"v1/runs/lineage",
```
Comment (dolfinus) [inline tests/test_server/test_lineage/test_get_lineage_request_validators.py:171]:
```suggestion
        "v1/runs/lineage",
```

File: data_rentgen/server/schemas/v1/lineage.py (hunk 1 above)
Comment (TiGrib) [reply data_rentgen/server/schemas/v1/lineage.py:82]: Mb `started_point_id` or `started_node_id`

Or you prefer without `point`?

Comment (dolfinus) [reply data_rentgen/server/schemas/v1/lineage.py:82]: Let's use `start_node_id`

**Review by dolfinus — APPROVED** [review]

**Pull request merged**